integration-tests/chain-signatures/tests/actions/wait_for.rs

```diff
@@ -238,30 +273,63 @@ pub async fn signature_responded(
         Ok(signature)
     };
 
-    let signature = is_tx_ready
-        .retry(&ExponentialBuilder::default().with_max_times(6))
-        .await
-        .with_context(|| "failed to wait for signature response")?;
-    Ok(signature)
+    let mut result: Result<FullSignature<Secp256k1>, WaitForError> = Err(
+        WaitForError::Signature(SignatureError::NotYetAvailable),
+    );
+
+    // retry getting tx status for 5 times
+    let mut retries = 5;
+    while let Err(error) = result {
+        match error {
+            WaitForError::Signature(SignatureError::Failed(_)) => return Err(error),
+            _ => {
+                if retries < 0 {
+                    return Err(error);
+                }
+                if retries < 5 {
+                    tokio::time::sleep(Duration::from_secs(20)).await;
+                }
+                result = is_tx_ready().await;
+            }
+        }
+        retries -= 1;
+    }
```

Comment:
We don't have to build our own custom retrying mechanism here. We can use `ConstantBuilder` for this:
```rs
    let strategy = ConstantBuilder::default()
        .with_delay(Duration::from_secs(20))
        .with_max_times(5);

    let signature = is_tx_ready
        .retry(&strategy)
        .await
        .with_context(|| "failed to wait for signature response")?;
```
and change the return type of `anyhow::Result<Outcome>` with:
```rs
enum Outcome {
    Signature(FullSignature<Secp256k1>),
    Failed,
}
```